### PR TITLE
fixes api definitions for reg/stack mixed calling i386 conventions. shou...

### PR DESCRIPTION
**Pull request by atlas0fd00m**
...ld not change symboliks.

this will be most noticeable in the gui rendering of arg-accessing opcodes.

### CODE REVIEW

File: vivisect/analysis/i386/calling.py [hunk 1]
```diff
@@ -113,13 +113,12 @@ def analyzeFunction(vw, fva):
     cc = emu.getCallingConvention(callconv)
     stcount = cc.getNumStackArgs(emu, argc)
     stackidx = argc - stcount
+    baseoff = cc.getStackArgOffset(emu, argc)
```
Comment (invisig0th) [inline vivisect/analysis/i386/calling.py]: Are there any calling conventions where this couldn't be a statically declared class-local value?  ( or we could probably make the default impl just return a declared class-local of 0 which things like msfastcall can declare as 12 or whatever? )

Comment (atlas0fd00m) [inline vivisect/analysis/i386/calling.py]: could.  i was just trying to make use of the value that calling conventions already provide with CC_STACK_INF, although it is less efficient.  i'll make the change and let you know.

**Comment by atlas0fd00m** [comment]
try again.  i didn't remove the call to cc.getStackArgOffset(), but rather improved it's performance by using "pad + align" to create a valid offset (correct for all current cconvs)

File: envi/__init__.py [hunk 2]
```diff
@@ -753,6 +753,9 @@ def getNumStackArgs(self, emu, argc):
         rargs = [ v for (t, v) in self.arg_def if t == CC_REG ]
         return max(argc - len(rargs), 0)
 
+    def getStackArgOffset(self, emu, argc):
```
Comment (l1berty) [inline envi/__init__.py]: add function docstring like, 'Returns offset to first stack argument'

add unittest for this new api to:
https://github.com/vivisect/vivisect/blob/master/vdb/testmods/callingconventions.py

File: envi/__init__.py [hunk 3]
```diff
@@ -753,6 +753,9 @@ def getNumStackArgs(self, emu, argc):
...
+        return self.pad + self.align    # one pointer for RET pointer and offset for CConv specialties
```
Comment (l1berty) [inline envi/__init__.py]: remove # comment

**Pull request merged**